packages/core/src/rules/utils.ts

```diff
@@ -164,3 +164,26 @@ export function validateSchemaEnumType(
     });
   }
 }
+
+export function validateResponseCodes(
+  responseCodes: string[],
+  codeRange: string,
+  { report }: UserContext
+) {
+  const responseCodeRegexp = new RegExp(`^${codeRange[0]}[0-9Xx]{2}$`);
+
+  const containsNeededCode = responseCodes.some((code) => {
+    if (code === 'default' && codeRange === '2XX') {
+      return true;
+    }
+
+    return responseCodeRegexp.test(code);
+  });
```

Comment:
This might be even better:

```suggestion
  const containsNeededCode = responseCodes.some((code) => 
    (codeRange === '2XX' && code === 'default') // It's OK to replace 2xx codes with the default
    || responseCodeRegexp.test(code));
```